js/app/viewmodel.js

```diff
@@ -45,25 +45,42 @@ class CompletionManager {
     this.rx_pinned.onNext(this.pinned);
   }
 
+  validateArray(newCompletion){
+    if (!Array.isArray(newCompletion)) {
+      window.alert("Error: Invalid fishing checklist.");
+      return false;
+    }else{
+      return true;
+    }
+  }
+
   importCaughtState() {
     var completion = window.prompt("Enter Fishing Checklist Code:");
     if (completion === null || completion === "") { return; }
     var newCompletion;
     try {
       newCompletion = JSON.parse(completion);
-      newCompletion = _(newCompletion).reduce((o, v) => o.concat(Number(v)), [])
     } catch(e) {
       window.alert("Error: Malformed fishing checklist.");
       return;
     }
-    if (!Array.isArray(newCompletion)) {
-      window.alert("Error: Invalid fishing checklist.");
-      return;
+    if (newCompletion["pinned"].length > 0) {
+      this.pinned = _(newCompletion["pinned"]).reduce((o, v) => o.concat(Number(v)), []);
+      if(this.validateArray(this.pinned)){
```

Comment:
Same as above for coding style.
```suggestion
      if (this.validateArray(this.pinned)) {
```